internal/armada/server/submit.go

```diff
@@ -59,41 +58,97 @@ func NewSubmitServer(
-	handler := handlers.CreateQueue(server.queueRepository.CreateQueue).
-		Ownership(authorization.GetPrincipal(ctx).GetName()).
-		Validate().
-		Authorize(server.permissions.UserHasPermission, permissions.CreateQueue)
+func (server *SubmitServer) CreateQueue(ctx context.Context, queue *api.Queue) (*types.Empty, error) {
+	if e := checkPermission(server.permissions, ctx, permissions.CreateQueue); e != nil {
+		return nil, e
+	}
+
+	if len(queue.UserOwners) == 0 {
+		principal := authorization.GetPrincipal(ctx)
+		queue.UserOwners = []string{principal.GetName()}
+	}
+
+	if err := validateQueue(queue); err != nil {
+		return nil, status.Errorf(codes.InvalidArgument, "[UpdateQueue] error: %s", err)
+	}
 
-	return handler(ctx, req)
+	e := server.queueRepository.CreateQueue(queue)
+	if e == repository.ErrQueueAlreadyExists {
+		return nil, status.Errorf(codes.AlreadyExists, "Queue %q already exists", queue.Name)
+	} else if e != nil {
+		return nil, status.Errorf(codes.Unavailable, e.Error())
+	}
+	return &types.Empty{}, nil
 }
 
-func (server *SubmitServer) UpdateQueue(ctx context.Context, req *api.Queue) (*types.Empty, error) {
-	handler := handlers.UpdateQueue(server.queueRepository.UpdateQueue).
-		Validate().
-		Authorize(server.permissions.UserHasPermission, permissions.CreateQueue)
+func (server *SubmitServer) UpdateQueue(ctx context.Context, queue *api.Queue) (*types.Empty, error) {
+	if e := checkPermission(server.permissions, ctx, permissions.CreateQueue); e != nil {
+		return nil, e
+	}
 
-	return handler(ctx, req)
+	if err := validateQueue(queue); err != nil {
+		return nil, status.Errorf(codes.InvalidArgument, "[UpdateQueue] error: %s", err)
+	}
+
+	e := server.queueRepository.UpdateQueue(queue)
+	if e == repository.ErrQueueNotFound {
+		return nil, status.Errorf(codes.NotFound, "Queue %q not found", queue.Name)
+	} else if e != nil {
+		return nil, status.Errorf(codes.Unavailable, e.Error())
+	}
+	return &types.Empty{}, nil
+}
+
+func validateQueue(queue *api.Queue) error {
+	if queue.PriorityFactor < 1.0 {
+		return fmt.Errorf("queue priority must be greater than or equal to 1, but is %f", queue.PriorityFactor)
+	}
+	return nil
 }
 
-func (server *SubmitServer) DeleteQueue(ctx context.Context, req *api.QueueDeleteRequest) (*types.Empty, error) {
-	handler := handlers.DeleteQueue(server.queueRepository.DeleteQueue).
-		CheckIfEmpty(server.jobRepository.GetQueueActiveJobSets).
-		Authorize(server.permissions.UserHasPermission, permissions.DeleteQueue)
+func (server *SubmitServer) DeleteQueue(ctx context.Context, request *api.QueueDeleteRequest) (*types.Empty, error) {
+	if e := checkPermission(server.permissions, ctx, permissions.DeleteQueue); e != nil {
+		return nil, e
+	}
+
+	active, e := server.jobRepository.GetQueueActiveJobSets(request.Name)
+	if e != nil {
+		return nil, status.Errorf(codes.InvalidArgument, e.Error())
+	}
+	if len(active) > 0 {
+		return nil, status.Errorf(codes.FailedPrecondition, "Queue is not empty.")
```

Comment:
is there no way to put the name of the queue here?

Reply:
I'm sure there is, but it may be cleaner to do that in a separate PR. So that this PR is only about reverting a prior change. For example, the following one.
https://github.com/G-Research/armada/pull/829